src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/PrefetchGetRecordsCache.java

```diff
@@ -177,8 +177,11 @@ public void run() {
                         
                         MetricsHelper.getMetricsScope().addData(EXPIRED_ITERATOR_METRIC, 1, StandardUnit.Count,
                                 MetricsLevel.SUMMARY);
-                        
-                        dataFetcher.restartIterator();
+                        try {
+                            dataFetcher.restartIterator();
+                        } catch (Throwable ex) {
+                            log.error("Exception thrown while restarting iterator", ex);
```

Comment:
Why are we suppressing now?

Reply:
The existing error handling in this level is to catch exception and log errors. Do we wanna handle this error differently?

Reply:
It is already handled as part of outer try. We should be good.